cli/src/clojure_lsp/server.clj

```diff
@@ -181,6 +183,37 @@
   (getWorkspaceService [_]
     (LSPWorkspaceService. handler)))
 
+(defrecord ClojureLspProducer [lsp-producer ^ClojureLanguageClient client db]
+  producer/IProducer
+  (publish-diagnostic [_this diagnostic]
+    (producer/publish-diagnostic lsp-producer diagnostic))
+  (refresh-code-lens [_this]
+    (producer/refresh-code-lens lsp-producer))
+  (publish-workspace-edit [_this edit]
+    (producer/publish-workspace-edit lsp-producer edit))
+  (show-document-request [_this document-request]
+    (producer/show-document-request lsp-producer document-request))
+  (publish-progress [_this percentage message progress-token]
+    (producer/publish-progress lsp-producer percentage message progress-token))
+  (show-message-request [_this message type actions]
+    (producer/show-message-request lsp-producer message type actions))
+  (show-message [_this message type extra]
+    (producer/show-message lsp-producer message type extra))
+  (register-capability [_this capability]
+    (producer/register-capability lsp-producer capability))
+
+  producer/IClojureProducer
+  (refresh-test-tree [_this uris]
+    (go
+      (when (some-> @db/db :client-capabilities :experimental j/from-java :testTree)
+        (shared/logging-time
+          "Refreshing testTree took %s secs"
+          (doseq [uri uris]
+            (when-let [test-tree (f.test-tree/tree uri db)]
+              (->> test-tree
+                   (coercer/conform-or-log ::coercer/publish-test-tree-params)
+                   (.publishTestTree client)))))))))
+
```

Comment:
I'm ok with that, but what's the difference between IClojureProducer and IProducer?

Reply:
IProducer has methods that can have a generic implementation in lsp4clj, while IClojureProducer holds Clojure specific methods.
I'll have to do the same thing with clojure-lsp.handler/IHandler.

Reply:
Got it, so the idea is that one could use lsp4clj for other servers than clojure ones right?

Reply:
Yeah, like borkdudes zen or in my case an internal language at work.
Right now it would be kind of a wrapper around lsp4j, but maybe we could throw out lsp4j at some point.
If we're crazy enough it might even become a cljc project so that it can run in the browser, but that's a ways off 😄